L1T online DQM HLT path for Heavy Ion

BetterWang opened the pull request:
[+] adding HIPhysics to the HLT

christopheralanwest committed:
Implement generation of LUTs for HE FG bits

eioannou committed:
remove bad pf candidates clustered to jets

eioannou committed:
remove bad pf candidates clustered to jets

kpedro88 committed:
cleanup

kpedro88 committed:
simplify

kpedro88 committed:
respect user choice in reclusterJets

kpedro88 committed:
add fixEE2017 and fixEE2017Params to MET tool

kpedro88 committed:
make separate good jet collection

kpedro88 committed:
modules for EE fix in MET tool

kpedro88 committed:
fix params

kpedro88 committed:
get task

kpedro88 committed:
fix jet type

kpedro88 committed:
avoid product confusion

kpedro88 committed:
improve default settings

kpedro88 committed:
add to sequence

kpedro88 committed:
remove unnecessary default

kpedro88 committed:
use correct collection in updateJECs()

kpedro88 committed:
use vector of Ptr instead of PtrVector (multithreading issue)

kpedro88 committed:
add switch to disable MET significance computation

makortel committed:
Use SimHits instead of digis for muons

This is closer to what is done in classical mixing, and turns out to
lead slightly smaller file sizes as well.

christopheralanwest committed:
Get HE TP FG version from database

christopheralanwest committed:
Fix behavior of HE depth feature bit algorithm

christopheralanwest committed:
Update LUT generation template for 2018

christopheralanwest committed:
Add new HE fine grain bit algorithms

kpedro88 committed:
actually veto the superbad candidates

christopheralanwest committed:
Generate FG LUTs only for physical HT channels in HE

christopheralanwest committed:
Needed changes to HCAL LUT loader conventions

christopheralanwest committed:
Apply code checks

apsallid committed:
Update rechit calibration: dEdX weights, fCPerMIP, thickness corrections

Martin-Grunewald committed:
Cleanup and 2018 HIon HLT menu

eioannou committed:
Change of Pt cut from 75 to 50GeV

christopheralanwest committed:
Migrate from boost::shared_ptr to std::shared_ptr

christopheralanwest committed:
Apply code checks

hroskes committed:
fix conddb function in 10_3

apsallid committed:
assign fCPerMIP to a variable

Commit:
Add the possibility of navigation in HGCalTopology

makortel committed:
Fix GEM validation for phase2 premixing

These parameters point to "real digis" instead of "sim digis", so they
should not be customized for premixing.

christopheralanwest committed:
Merge branch 'master' into hcal-generate-he-fg-lut-new-algo-versions

Commit:
Changed to enum

Commit:
Adding new folder to test new readouts for RPC DQM online

Commit:
cleaning

cardinia committed:
Implemented option to run jobs in parallel for PV validation

wddgit committed:
Remove const_cast from SiPixelESProducers

Note these const_cast's were necessary before
PR #24844 was merged.

bundocka committed:
Fix caloParamsHelper conflict

cardinia committed:
Update primaryVertexValidation.py

eioannou committed:
change to lower case initials

eioannou committed:
change to lower case initials

eioannou committed:
change to lower case initials

Commit:
Treat the subdetid method correctly

makortel committed:
Move HGCal event content customizations to SimCalorimetry

makortel committed:
Move MTD event content customizations to SimFastTiming

bundocka committed:
Fix LUT depths

mmusich committed:
fix testProduceSystematicMisalignment and add unit tests

bundocka committed:
Update utilities

bundocka committed:
ChunkySandwich -> PhiRing

bundocka committed:
Merge branch 'master' into furtherO2OUpdates_mast_v3

wddgit committed:
Prepare for concurrent IOVs, L1Trigger

Change the return type of ESProducer produce methods
from shared_ptr to unique_ptr.

fwyzard committed:
Skip tests if no CUDA devices are present

If no CUDA devices are present, skip the tests and return a success
status, instead of failing.

fwyzard committed:
Fix the check in case a CUDA driver is not available

fwyzard committed:
Improve error message

eioannou committed:
Change of Pt cut from 75 to 50GeV

eioannou committed:
Change of Pt cut from 75 to 50GeV

muhammadansariqbal committed:
Migration from user based output EOS directories to common output EOS directory for Tracker Alignment Validation.

muhammadansariqbal committed:
Tracker alignment validation, made eosdir a mandatory input to avoid outputs being overwritten owing to default names.

fabiocos committed:
Test for geometry, fix to TimingSD class, work in progress

fabiocos committed:
Fix post-step local point calculation, reset the time slice unit to 10 ps, switch off verbosity

schneiml committed:
Fix caching logic in SiStripBadComponentInfo

... by removing it. Also avoid a needless copy.

pieterdavid committed:
Fix APV shot cleaner for 10bit (if any digi is >255)

smuzaffar committed:
use ofast-flag tool instead of directly CXXFLAGS=-Ofast

Commit:
changing name 'RPCTwinMuxRawToDigi' to 'rpcTwinMuxRawToDigi' and adding 'rpcCPPFRawToDigi' for the HCAL and L1T

Commit:
Update material property as discussed in DPG

mandrenguyen committed:
Fixes to EI and validation for running on data with the Run2_2018_pp_on_AA era

mandrenguyen committed:
Remove spurious line add

schneiml committed:
Reset _vhashCrates for each run.

Seems to be done for the other vectors as well, and else crashes in multi-run harvesting.

argiro committed:
trivial bug fixes

Commit:
fixed wrong deleting

mmusich committed:
add missing eventSetupPathsKey swap

cedricpri committed:
Moved to global muon web repository

emacdonald16 committed:
Added new module for stub pt consistency

BetterWang committed:
[+] adding HIPhysics to the HLT

cmsbuild commented:
The code-checks are being triggered in jenkins.

andrius-k commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

thomreis commented on DQM/Integration/python/clients/l1tstage2emulator_dqm_sourceclient-live_cfg.py:
```diff
@@ -65,7 +65,7 @@
 process.hltFatEventFilter.throw = cms.bool(False)
 # HLT_Physics now has the event % 107 filter as well as L1FatEvents
-process.hltFatEventFilter.HLTPaths = cms.vstring('HLT_L1FatEvents_v*', 'HLT_Physics_v*')
+process.hltFatEventFilter.HLTPaths = cms.vstring('HLT_L1FatEvents_v*', 'HLT_Physics_v*', 'HLT_HIPhysics_v*')
```
I think it would be better to change this in the `if (process.runType.getRunType() == process.runType.hi_run):` block.

fabiocos replied:
@BetterWang @andrius-k was this comment answered/addressed?

cmsbuild committed:
Merge pull request #24914 from wddgit/removeConstCastSiPixel

Remove const_cast from SiPixelESProducers

Commit:
Take Yana's comments

emacdonald16 committed:
Addressing Kevin's suggestions

Commit:
Fixing Bug and adding missing collections for HI Run

emacdonald16 committed:
Still addressing one of Kevin's comments

cmsbuild commented:
+1
Tested at: 9230af59db868d5e2ac45ad0cdf88348e71db2b4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24973/31225/summary.html

cmsbuild commented:
Comparison job queued.

efeyazgan committed:
adding CP5 hadronizers for relvals

efeyazgan committed:
updating hadronizer locations to have CP5

mdhildreth committed:
fix memory leak

Commit:
adding rpcTwinMuxRawToDigi for HI Run

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24973/31225/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2994843
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2994645
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 134 log files, 14 edm output root files, 32 DQM output files

cmsbuild committed:
Merge pull request #24952 from muhammadansariqbal/CMSSW_10_3_X_EOSMigration

Tracker Alignment - Fix to conddb function in validation and migration to common EOS storage

cmsbuild committed:
Merge pull request #24938 from fwyzard/skip_CUDA_tests_104x

Skip tests if no CUDA devices are present

cmsbuild committed:
Merge pull request #24931 from bundocka/furtherO2OUpdates_mast_v3

Further CaloL2 O2O updates (supersedes #24922)

cmsbuild committed:
Merge pull request #24906 from cardinia/patch-1

Implemented option to run jobs in parallel for PV validation

cmsbuild committed:
Merge pull request #24876 from cms-met/METFixEE2017_949_v2_portTomaster

MET EE noise mitigation in 2017 data

cmsbuild committed:
Merge pull request #24929 from makortel/eventContentReorg

Move HGCal and MTD event content customizations to SimCalorimetry and SimFastTiming

cmsbuild committed:
Merge pull request #24955 from schneiml/dqm-fix-sistripbadcomponent-caching

SiStripMonitorClient: Fix caching logic in SiStripBadComponentInfo

cmsbuild committed:
Merge pull request #24556 from christopheralanwest/hcal-generate-he-fg-lut-new-algo-versions

Complete HE depth feature bit implementation

cmsbuild committed:
Merge pull request #24963 from argiro/EcalSampleMaskFix

Trivial bug fixes to record  EcalSampleMask

cmsbuild committed:
Merge pull request #24975 from mmusich/customizeSiStripCalMinBiasAAG_For_pp_on_AA

[10.4.X] add missing eventSetupPathsKey swap for HI 2018

smuzaffar committed:
Added lxml and bs4 in python unit tests

This needs https://github.com/cms-sw/cmsdist/pull/4447 externals

cmsbuild committed:
Merge pull request #24882 from apsallid/rechitcalibhgcalv9

Update rechit calibration: dEdX weights, fCPerMIP, thickness corrections

efeyazgan committed:
add hadronizers for tune CP5

mandrenguyen committed:
added goodOfflinePV to front of EI sequence

smuzaffar committed:
do not build/run cuda tests if cuda does not support gcc version

smuzaffar committed:
Update BuildFile.xml

gartung committed:
Move postprocess-scan-build.py to Utilities/StaticAnalyzers

casarsa committed:
Fixed the ADC and TDC saturation

wddgit committed:
Delete unused using directives, cleanup

cmsbuild committed:
Merge pull request #24971 from cedricpri/new-muon-web

Moved to new muon validation web repository

cmsbuild committed:
Merge pull request #24959 from schneiml/dqm-multirunharvesting-fix-hcal

HCAL DQClient: Reset _vhashCrates for each run.

emacdonald16 committed:
Changed speed of light to CLHEP::c_light, and made SOL and unit conversion a single constant

Commit:
Enable saving Digi results of HFNose and instead of PR #24913

apana committed:
Update uGT unpacker to pick up new HI centrality values

cmsbuild committed:
Merge pull request #24980 from mdhildreth/MemLeakFixMaster

fix memory leak

cmsbuild committed:
Merge pull request #24925 from mmusich/fixMisalignmentToolsPlusTests_104X

[10.4.X] Fix testProduceSystematicMisalignment and add unit tests

cmsbuild committed:
Merge pull request #24957 from bsunanda/Phase2-hgx163

Phase2-hgx163 Update material property as discussed in DPG

efeyazgan committed:
Merged relvalCP5 from repository efeyazgan with cms-merge-topic

cmsbuild committed:
Merge pull request #24958 from mandrenguyen/valFix4Prompt_ppOnAA_104X

Bug fixes to Run2_2018_pp_on_AA in EI and DQM

smuzaffar committed:
drop architecture amd64 check for cuda tests

Commit:
Fixed JetMET summary plot for HI DQM

cmsbuild committed:
Merge pull request #24961 from pieterdavid/sistripapvshotcleaner10bit

Fix APV shot cleaner for 10bit

cmsbuild committed:
Merge pull request #24954 from fabiocos/fc-fixGeom

MTD: fix TimingSD calculation of post-step local point and time slice unit definition, add SD geometry dump module

Commit:
Apply code checks

cmsbuild committed:
Merge pull request #24897 from bsunanda/Phase2-hgx162

Phase2-hgx162 Add the possibility of navigation in HGCalTopology

mmusich committed:
customized the DQM tracking only sequence when run in trackingOnlyEra_Run2_2018_pp_on_AA to remove hiConformalPixelTracks

cmsbuild committed:
Merge pull request #24985 from bsunanda/Phase2-hgx164

Phase2-hgx164 Enable saving Digi results of HFNose and instead of PR #24913

gartung committed:
use command line input

efeyazgan committed:
one more missing hadronizer for CP5 added

Commit:
use ofast-flag instead of -Ofast CXXFLAGS

mandrenguyen committed:
Add pixel tracks to XeXe era

cmsbuild committed:
Merge pull request #24902 from slezki/RPCDQMNewReadouts_1030pre6

New folder in rpcdqm online to test new readouts

Commit:
Added pixel activity filter to EventDisplay

Commit:
Added pixel activity filter to Beam Monitor

tocheng committed:
Using pp_on_AA_2018 era to configure DtCalib AlCaReco the same as DtCalibHI

cmsbuild committed:
Merge pull request #24986 from apana/apana-AddCentralityToUnpacker_CMSSW_10_4_X

Update uGT unpacker to pick up new HI centrality values

zhenhu committed:
add 2 production-like workflows for HI 2018

alberto-sanchez committed:
Merged relvalCP5 from repository efeyazgan with cms-merge-topic

tocheng committed:
Using pp_on_AA_2018 era to configure HcalMinBias AlCaReco the same as HcalMinBiasHI

tocheng committed:
Add offlinePrimaryVertices instead of hiSelectedVertex

cmsbuild committed:
Merge pull request #25000 from mandrenguyen/hiConfPixelTracksXeXe104X

Add pixel tracks to XeXe era

slava77 committed:
streamline addition of centrality and pixel tracks for HI

cmsbuild committed:
Merge pull request #25004 from tocheng/DtCalib_CMSSW_10_4_X

Configure DtCalib AlCaReco for heavy Ion 2018

cmsbuild committed:
Merge pull request #24996 from mmusich/customiseDQMSeq_forHiConformalTracksInTrackingOnly_103X

[10.4.X] customize the DQM tracking only sequence when run in trackingOnlyEra_Run2_2018_pp_on_AA to remove hiConformalPixelTracks

cmsbuild committed:
Merge pull request #25007 from tocheng/HcalCalMinBias_CMSSW_10_4_X

Configure HcalMinBias AlCaReco for heavy Ion 2018

cmsbuild committed:
Merge pull request #25011 from slava77/patch-44

streamline addition of centrality and pixel tracks for HI

Commit:
Disabling Event Display and Beam Spot filters used in 2018 HPU run

smuzaffar committed:
Remove the generated report file for the duplicate entries

Currently we have 110K+ entries  and report files. Although after removing the duplicates we are left with 5K+ entries in the index.html but this change should also make sure to deleted the duplicate report-xxxx.html files.

smuzaffar committed:
decompse the row after the deletion of report files

zhenhu committed:
update input dataset names

civanch committed:
removed dependency violations

makortel committed:
Remove unnecessary casts from Alignment/CocoaToDDL/src/CocoaUnitsTable.cc

makortel committed:
Remove const-away cast from OnlineDB/EcalCondDB/src/LMFColor.cc

makortel committed:
Remove const-away cast from TopQuarkAnalysis/TopEventProducers/src/TopDecaySubset.cc

cmsbuild committed:
Merge pull request #24917 from bsunanda/Run2-hcx186

Run2-hcx186 Treat the subdetid method correctly

cmsbuild committed:
Merge pull request #24827 from makortel/premixingMuonSimHits

Use SimHits instead of digis for muon detectors in premixing

cmsbuild committed:
Merge pull request #24991 from casarsa/mc-simBTL-fix

MTD simulation: fix to the BTL digitization

makortel committed:
Remove const-away cast from CalibCalorimetry/EcalCorrelatedNoiseAnalysisAlgos/src/TEcnaGui.cc

makortel committed:
Remove const-away cast from CalibTracker/SiStripHitEfficiency/src/TrajectoryAtInvalidHit.cc

makortel committed:
Remove const-away cast from OnlineDB/EcalCondDB/src/LMFDefFabric.cc

makortel committed:
Remove const-away cast from OnlineDB/EcalCondDB/interface/IODConfig.h

makortel committed:
Remove const-away cast from SimTracker/TrackerMaterialAnalysis/plugins/TrackingMaterialProducer.cc

makortel committed:
Remove const-away cast from AnalysisAlgos/TrackInfoProducer/src/TrackInfoProducerAlgorithm.cc

makortel committed:
Remove const-away cast from TopQuarkAnalysis/TopHitFit/src/PatJetHitFitTranslator.cc

makortel committed:
Remove const-away cast from SimG4CMS/ShowerLibraryProducer/src/FiberSD.cc

makortel committed:
Remove const-away cast from MuonAnalysis/MomentumScaleCalibration/plugins/MuScleFitMuonSelector.cc

makortel committed:
Remove const-away cast from SimG4CMS/Muon/src/MuonSensitiveDetector.cc

makortel committed:
Remove const-away cast from SimG4Core/SensitiveDetector/src/SensitiveDetector.cc

makortel committed:
Remove const-away cast from SimG4Core/Application/src/GFlashEMShowerModel.cc

smuzaffar committed:
Merge pull request #24982 from cms-sw/smuzaffar-patch-2

Added lxml and bs4 in python unit tests

cmsbuild committed:
Merge pull request #24969 from smuzaffar/from-CMSSW_10_4_CLANG_X_2018-10-21-2300

use ofast-flag tool instead of directly CXXFLAGS=-Ofast

cmsbuild committed:
Merge pull request #24983 from smuzaffar/unitetss-cuda-gcc-support

Do not build/run cuda tests if cuda does not support gcc version

cmsbuild committed:
Merge pull request #24984 from gartung/Utilites-StaticAnalyzers-postproc

Add postprocess-scan-build.py to Utilities/StaticAnalyzers

cmsbuild committed:
Merge pull request #25005 from zhenhu/addProductionlikeWorkflowHI2018

add 2 production-like workflows for HI 2018

cmsbuild committed:
Merge pull request #24972 from emacdonald16/L1TriggerSPC

L1 trigger - TTTrack - stub pt consistency

andrius-k commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild committed:
Merge pull request #25028 from civanch/removed_digi_dependence_on_hits

Removed dependency violations

makortel committed:
Code checks

makortel committed:
Remove catch(...) from MuonAnalysis/MomentumScaleCalibration/plugins/MuScleFitMuonSelector.cc

smuzaffar committed:
run cx_oracle only if tool if available e.g. only amd64

makortel committed:
Remove const-away cast from OnlineDB/EcalCondDB/src/MODCCSHFDat.cc

cmsbuild committed:
Merge pull request #25024 from andrius-k/Removing_BeamSpot_EventDisplay_Filters

Disabling BeamSpot EventDisplay filters

cmsbuild committed:
Merge pull request #25013 from alberto-sanchez/efe-cp5

(once again) adding cp5 for relvals

cmsbuild committed:
Merge pull request #24951 from wddgit/concurrentIOVsL1Trigger2

Prepare for concurrent IOVs, L1Trigger

cmsbuild committed:
Merge pull request #25038 from makortel/saConstCastAlca

Remove some const-away casts from AlCa

cmsbuild committed:
Merge pull request #25039 from makortel/saConstCastAnalysis

Remove some const-away casts from analysis

cmsbuild committed:
Merge pull request #25045 from makortel/saConstCastSim

Remove some const-away casts from simulation

cmsbuild committed:
Merge pull request #25046 from smuzaffar/cx_oracle-unittest

run cx_oracle only if tool if available e.g. only amd64

BetterWang committed:
[!] update runType

cmsbuild commented:
The code-checks are being triggered in jenkins.

BetterWang commented:
Hi @fabiocos , sorry for missing the comment. Just updated the PR.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24973/7034

cmsbuild commented:
Pull request #24973 was updated. @kmaeshima, @cmsbuild, @andrius-k, @jfernan2, @schneiml can you please check and sign again.

cmsbuild committed:
Merge pull request #25029 from cms-tsg-storm/HLT103XDevelopment

2018 HIon HLT menu (104X)

andrius-k commented:
This PR has conflicting file. @BetterWang could you please take a look at this?

cmsbuild committed:
Merge pull request #24993 from jusaviin/jetDQMSummaryHI

Fixed JetMET summary plot for HI DQM

cmsbuild committed:
Merge pull request #25041 from makortel/saConstCastDB

Remove some const-away casts from DB

BetterWang committed:
Merge branch 'L1T_103X_HItest' of github.com:BetterWang/cmssw into L1T_103X_HItest

cmsbuild commented:
The code-checks are being triggered in jenkins.

slava77 commented:
-1

wrong branch, apparently

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24973/7043

cmsbuild commented:
Pull request #24973 was updated. @andrius-k, @lveldere, @schneiml, @ianna, @kpedro88, @nsmith-, @rekovic, @thomreis, @alberto-sanchez, @pohsun, @peruzzim, @civanch, @zhenhu, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @efeyazgan, @mdhildreth, @jfernan2, @tocheng, @ggovi, @qliphy, @fabiocos, @ssekmen, @Martin-Grunewald, @prebello, @kmaeshima, @perrozzi, @monttj, @pgunnell, @franzoni, @fgolf, @lpernie can you please check and sign again.
